flake.nix

```
@@ -71,6 +71,15 @@
           inputMap = {
             "https://input-output-hk.github.io/cardano-haskell-packages" = inputs.CHaP;
           };
+
+          # force LANG to be UTF-8, otherwise GHC might choke on UTF encoded data.
+          shell.shellHook = ''
+            export LANG=en_US.UTF-8
+            export LC_ALL=en_US.UTF-8
+          '' + lib.optionalString (nixpkgs.glibcLocales != null && nixpkgs.stdenv.hostPlatform.libc == "glibc") ''
+            export LOCALE_ARCHIVE="${nixpkgs.glibcLocales}/lib/locale/locale-archive"
```

setting `LOCALE_ARCHIVE` to use different version of glibc that is used by the native system [might break native applications](https://github.com/NixOS/nix/issues/4829#issuecomment-844376944) so probably safer to use `LOCALE_ARCHIVE_2_37`:
```suggestion
            export LOCALE_ARCHIVE_${builtins.replaceStrings ["."] ["_"] nixpkgs.glibcLocales.version}="${nixpkgs.glibcLocales}/lib/locale/locale-archive"
```

---

Unfortunately that suggestion only caused problems and brought back the original issue.